### PR TITLE
Ask only for VAT ID if company is inside EU

### PR DESCRIPTION
**Pull request by ateuber**
To avoid confusion in a reverse charge setting we don't want to ask companies that are not in the European Union for the VAD ID. Maybe it would be better, to make this a setting, but I don't have a good idea how to do it.

### CODE REVIEW

File: src/pretix/base/forms/questions.py [hunk 1]
```diff
@@ -630,6 +630,10 @@ def __init__(self, *args, **kwargs):
         )
         self.fields['state'].widget.is_required = True
 
+        if cc and cc not in EU_COUNTRIES and fprefix + 'vat_id' in self.data:
+            self.data = self.data.copy()
+            del self.data[fprefix + 'vat_id']
```
Comment (raphaelm) [inline src/pretix/base/forms/questions.py:635]: Modifying ``self.data`` doesn't feel clean, I think the part in the ``clean()`` should suffice?

File: src/pretix/static/pretixcontrol/js/ui/main.js [hunk 2]
```diff
@@ -314,6 +314,31 @@ var form_handlers = function (el) {
...
+                        } else {
+                            dependent.closest(".form-group").hide();
+                        }
+                        dependent.prop("disabled", false);
+                        dependency_country.closest(".form-group").find("label .fa-spin").remove();
+                    });
+                }
+            };
+        dependency_country.on("change", update);
+        dependency_id_is_business_1.on("change", update);
```
Comment (raphaelm) [inline src/pretix/static/pretixcontrol/js/ui/main.js:339]: We probably also need one ``update();`` call to evaluate it after page load (i.e. if something is pre-set)

File: src/pretix/static/pretixcontrol/js/ui/main.js [hunk 3]
```diff
@@ -314,6 +314,31 @@ var form_handlers = function (el) {
...
+        var dependent = $(this),
+            dependency_country = $(this).closest(".panel-body, form").find('select[name$=country]'),
+            dependency_id_is_business_1 = $(this).closest(".panel-body, form").find('input[id$=id_is_business_1]'),
+            update = function (ev) {
+                if (dependency_id_is_business_1.length && !dependency_id_is_business_1.prop("checked")) {
+                    dependent.closest(".form-group").hide();
+                } else {
+                    dependent.prop("disabled", true);
+                    dependency_country.closest(".form-group").find("label").prepend("<span class='fa fa-cog fa-spin'></span> ");
+                    $.getJSON('/js_helpers/is_country_in_eu/?country=' + dependency_country.val(), function (data) {
```
Comment (raphaelm) [inline src/pretix/static/pretixcontrol/js/ui/main.js:327]: An AJAX call to get a list of 27 two-letter country codes feels like overkill. Maybe we can just pass it to JS as a ``data-`` attribute to the form field or something?

File: src/pretix/base/forms/questions.py (hunk 1 above)
Comment (ateuber) [reply src/pretix/base/forms/questions.py:635]: Not important, but the idea is to remove the input if JavaScript is disabled and the form is reinitialized with a selecte country that is outside the EU, similar to https://github.com/pretix/pretix/commit/b0b0f7474d7999a088444e0f1c0649f18397585d
grafik.

File: src/pretix/static/pretixcontrol/js/ui/main.js (hunk 2 above)
Comment (ateuber) [reply src/pretix/static/pretixcontrol/js/ui/main.js:339]:
```suggestion
        dependency_id_is_business_1.on("change", update);
        $(window).on("load", update);
```

File: src/pretix/static/pretixcontrol/js/ui/main.js (hunk 3 above)
Comment (ateuber) [reply src/pretix/static/pretixcontrol/js/ui/main.js]: You are right, it is now passed as a `data-` attribute.

File: src/pretix/static/pretixcontrol/js/ui/main.js (hunk 2 above)
Comment (raphaelm) [reply src/pretix/static/pretixcontrol/js/ui/main.js:331]: No, ``load`` is pretty late. ``load`` only happens after *all* resources on the page have been loaded, which can take a while. You can just call ``update();`` – we're in a ``$(function() { … })`` block and jQuery ensures the DOM is ready.

File: src/pretix/base/forms/questions.py (hunk 1 above)
Comment (raphaelm) [reply src/pretix/base/forms/questions.py:635]: Ok, got it. Maybe add a small comment for future readers? :)

Comment (ateuber) [reply src/pretix/base/forms/questions.py:636]: Thanks, I added a comment.

File: src/pretix/static/pretixcontrol/js/ui/main.js (hunk 2 above)
Comment (ateuber) [reply src/pretix/static/pretixcontrol/js/ui/main.js:332]: Thanks for the explanation, changed it.

File: src/pretix/static/pretixcontrol/js/ui/main.js [hunk 4]
```diff
@@ -314,6 +314,24 @@ var form_handlers = function (el) {
         dependency.closest('.form-group').find('input[name=' + dependency.attr("name") + ']').on("dp.change", update);
     });
 
+    $("input[name$=vat_id][data-countries-in-eu]").each(function () {
```
Comment (raphaelm) [inline src/pretix/static/pretixcontrol/js/ui/main.js:317]: This selector actually broke a form in the invoice settings, so I refined it a little

**Pull request merged**